### PR TITLE
ET-3584 add maintenance function

### PR DESCRIPTION
**Pull request by joecks**
## What 🕵️ 🔍

- adds elasticsearch and postgres provisioner

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/ET-3584)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: aws/b2b_personalization/_env/tenant/create_database.hcl [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  tenant_config = read_terragrunt_config(find_in_parent_folders("tenant_config.hcl"))
+}
+
+terraform {
+  source = "../../../../modules//lambda/maintenance/create_database"
+}
+
+inputs = {
+  tenant  = local.tenant_config.locals.name
+  command = "CreateDatabase"
```
Comment (joecks) [inline aws/b2b_personalization/_env/tenant/create_database.hcl:12]: "command" can be removed

File: aws/b2b_personalization/_env/tenant/documents_api.hcl [hunk 2]
```diff
@@ -60,6 +60,28 @@ dependency "postgres" {
...
+  mock_outputs_allowed_terraform_commands = ["validate", "providers"]
+}
+
+dependency "create_database" {
+  config_path = "../create_database"
+
+  mock_outputs = {
+    result_success     = true
+    result_status_code = "200"
+    result_body        = "FAKE DATABASE SUCCESS"
```
Comment (joecks) [inline aws/b2b_personalization/_env/tenant/documents_api.hcl:80]: mock output is wrong currently, right now only success exists

File: aws/b2b_personalization/_env/tenant/users_api.hcl [hunk 3]
```diff
@@ -60,6 +60,28 @@ dependency "postgres" {
...
+
+dependency "create_database" {
+  config_path = "../create_database"
+
+  mock_outputs = {
+    result_success     = true
+    result_status_code = "200"
+    result_body        = "FAKE DATABASE SUCCESS"
+  }
+  mock_outputs_allowed_terraform_commands = ["validate", "providers"]
```
Comment (joecks) [inline aws/b2b_personalization/_env/tenant/users_api.hcl:82]: mock output is wrong currently, right now only success exists

File: aws/b2b_personalization/dev/tenants/xayn_test_2/tenant_config.hcl [hunk 4]
```diff
@@ -0,0 +1,3 @@
+locals {
+  name = "xaynTest2"
```
Comment (joecks) [inline aws/b2b_personalization/dev/tenants/xayn_test_2/tenant_config.hcl:2]: delete the second tenant at the end of this PR

**Review by jorge-cr-13 — APPROVED** [review]

**Pull request merged**